component/src/main/java/org/wso2/extension/siddhi/execution/env/GetYAMLProperty.java

```diff
@@ -0,0 +1,242 @@
+
+        if (attributeExpressionExecutors.length < 1) {
+            throw new SiddhiAppValidationException(
+                    "Invalid no of arguments passed to env:getYAMLProperty() function, " +
+                            "required at least 1, but found " + attributeExpressionExecutors.length);
+        }
+
+        Attribute.Type attribute1Type = attributeExpressionExecutors[0].getReturnType();
+        if (!((attribute1Type == Attribute.Type.STRING))) {
+            throw new SiddhiAppValidationException("Invalid parameter type found " +
+                    "for the argument Key of getYAMLProperty() function, " +
+                    "required " + Attribute.Type.STRING +
+                    ", but found " + attribute1Type.toString());
+        }
+
+        if (attributeExpressionExecutors.length > 1) {
+            if (!(attributeExpressionExecutors[1] instanceof ConstantExpressionExecutor)) {
+                throw new SiddhiAppValidationException("The second argument has to be a string constant specifying " +
+                        "one of the supported data types "
+                        + "(int, long, float, double, string, bool)");
+            } else {
+                String type = attributeExpressionExecutors[1].execute(null).toString();
+                if ("int".equals(type)) {
+                    returnType = Attribute.Type.INT;
+                } else if ("long".equalsIgnoreCase(type)) {
+                    returnType = Attribute.Type.LONG;
+                } else if ("float".equalsIgnoreCase(type)) {
+                    returnType = Attribute.Type.FLOAT;
+                } else if ("double".equalsIgnoreCase(type)) {
+                    returnType = Attribute.Type.DOUBLE;
+                } else if ("bool".equalsIgnoreCase(type)) {
+                    returnType = Attribute.Type.BOOL;
+                } else if ("string".equalsIgnoreCase(type)) {
+                    returnType = Attribute.Type.STRING;
+                } else {
+                    throw new SiddhiAppValidationException("Type must be one of int, long, float, double, bool, " +
+                            "string");
+                }
+            }
+        }
+
+        if (attributeExpressionExecutors.length > 2) {
+            Attribute.Type attribute3Type = attributeExpressionExecutors[2].getReturnType();
+
+            if (!(attribute3Type == returnType)) {
+                throw new SiddhiAppValidationException("Type of parameter default.Value " +
+                        "needs to match parameter data.type");
+            }
+        }
+
+        this.reader = reader;
+    }
+
+    /**
+     * The main execution method which will be called upon event arrival
+     * when there are more than one function parameter.
+     *
+     * @param data the runtime values of function parameters.
+     * @return the function result.
+     */
+    @Override
+    protected Object execute(Object[] data) {
+
+        String key = (String) data[0];
```

Comment:
Check the array length before accessing data. Otherwise, it will lead to ArrayIndexOutOfBoud exception.